### PR TITLE
Add single weather-representation endpoint

### PR DESCRIPTION
**Pull request by IvannikovG**
Implemented an endpoint for host/ weather-stations/{stationID}

If there is a station it will be returned as per specification for JSON and CSV. If there is no such station 404 is returned.
Includes two tests for JSON and CSV representation

### CODE REVIEW

File: handlers/data_access.js [hunk 1]
```diff
@@ -124,10 +125,79 @@ function getWeatherStations (stationCatalog) {
     })
   }
 }
-
-// TODO @Georgii implement
+// GET Single weather station
```
Comment (MoritzStueber) [inline handlers/data_access.js]: - empty line above comment on following block of code, not below
- also, pattern for these comments is VERB /path/:pathvar

File: sample_data/sample_public_key.pem [hunk 2]
```diff
@@ -0,0 +1,9 @@
+-----BEGIN PUBLIC KEY-----
```
Comment (MoritzStueber) [inline sample_data/sample_public_key.pem]: I moved this file to `auth/` previously, should only exist there (looks like an accidental commit?) -> delete here, including folder

File: handlers/data_access.js [hunk 3]
```diff
@@ -124,10 +125,79 @@ function getWeatherStations (stationCatalog) {
...
+      return url.format({
+        protocol: req.protocol,
+        host: req.get('host'),
+        pathname: 'weather-stations/' + station.stationId + '/' + parameter
+      })
+    }
+
+    const stations = su.findStationsInVicinityOf(stationCatalog, undefined, undefined, undefined)
+    const urlString = req.originalUrl
+    const stationId = urlString.split('/')[2]
```
Comment (MoritzStueber) [inline handlers/data_access.js]: all variables should be declared at the top of a function -- javascript does this anyway ("hoisting"), improves readability compared to mix of variable- and function declarations here

File: handlers/data_access.js [hunk 4]
```diff
@@ -124,10 +125,79 @@ function getWeatherStations (stationCatalog) {
...
+    if (station === undefined) {
+      res.set('Content-Type', 'application/problem+json')
+      res.status(404).send({
+        title: 'Not Found',
+        status: 404,
+        detail: 'The requested resource was not found on this server'
+      })
+      log.info('sent `404 Not Found` as response to ' + req.method + '-request on ' + req.path)
+      return
+    }
```
Comment (MoritzStueber) [inline handlers/data_access.js]: code duplication :anguished:  (fixed in my latest commits)

File: test/test_openapi_validation.js [hunk 5]
```diff
@@ -120,4 +120,24 @@ describe('Verify behaviour of API-instance against OAS and/or expectations', fun
...
+  describe('GET /weather-stations/10505', function () {
+    it('should return single JSON station representation', async function () {
+      const stationUrl = API_ORIGIN + '/weather-stations/10505'
+      const res = await axios.get(stationUrl, { headers: { Accept: 'application/json' } })
+
+      expect(res.status).to.equal(200)
+      assert.include(res.headers['content-type'], 'application/json')
+    })
+  })
+
```
Comment (MoritzStueber) [inline test/test_openapi_validation.js:133]: the crash you observed when requesting stationId 1050 (a non-existent one) would have been a good reason to write a test for that -> makes it reproducible, ensures that it is eventually fixed for good

**Review by MoritzStueber — COMMENTED** [review]
- works fine :+1: 
- duplicated code for sending 404 not such a good idea
- best practice is to declare variables at top of function
- you forgot to run the linter :wink: might enforce that later

I already fixed some of the issues (and marked them as resolved in this conversation)
=> please review all changes, remove the public key & we're good to go!

File: sample_data/sample_public_key.pem (hunk 2 above)
Comment (IvannikovG) [reply sample_data/sample_public_key.pem]: yes, correct

File: test/test_openapi_validation.js (hunk 5 above)
Comment (IvannikovG) [reply test/test_openapi_validation.js:133]: ok added

**Pull request merged**